bodhi/server/validators.py

```diff
@@ -989,6 +989,39 @@ def validate_comment_id(request, **kwargs):
         request.errors.status = HTTPNotFound.code
 
 
+def validate_new_comment(request, **kwargs):
+    """
+    Unflatten the content of new comment to accord with 'SaveCommentSchema' data structure.
+
+    For list of bug_feedback and testcase_feedback, the request object has data in format as:
+        "bug_feedback.0.bug_id": "1234", "bug_feedback.0.karma": "1"
+    which, in order for schema to be validated, needs to be converted as follows:
+        bug_feedback: [ { "bug_id": "1234", "karma": "1" }]
+
+    Args:
+        request (pyramid.util.Request): The current request.
+        kwargs (dict): The kwargs of the related service definition. Unused.
+    """
+    # In-method import used because there is mutual referencing between
+    # 'bodhi.server.schemas' and 'bodhi.server.validators'
+    from bodhi.server.schemas import SaveCommentSchema
+
+    content = str(request.body)
+
+    if content:
+        content = eval(content)
+
+        if type(content) is dict:
+            if 'bug_feedback' not in content and 'testcase_feedback' not in content:
+                content = SaveCommentSchema().unflatten(content)
+            content = str(content)
+
+    if type(content) is str:
```

Comment:
You should check the pyramid escape function for this. See https://docs.pylonsproject.org/projects/pyramid/en/latest/_modules/html.html#escape

Reply:
In the line you are suggesting, I am not escaping quotes - but instead converting single quotes to double quotes because that is the format colander parses string in.

However, it might be good to actually escape the ```text``` field of request to filter HTML characters.

Reply:
ok I need to look at colander in more details :small_airplane: